lib/reader.js

```diff
@@ -1,46 +1,25 @@
 'use strict';
 
-const pack = require('browser-pack');
-const getStream = require('get-stream');
 const assert = require('assert');
-
-function getRows(feeds) {
-    const existMap = new Map();
-    return feeds
-        .reduce((acc, feed, order) => {
-            feed.forEach(row => {
-                const copy = { ...row };
-                // add browserify.order and flatten
-                delete copy.order;
-                if (copy.entry === true) {
-                    copy.order = order;
-                }
-                acc.push(copy);
-            });
-            return acc;
-        }, [])
-        .filter(row => {
-            assert(
-                row,
-                `Expected feed entry to be an object, got "${typeof row}"`
-            );
-            assert(
-                row.id,
-                `Expected feed entry to have an id, got "${row.id}"`
-            );
-            if (existMap.has(row.id)) {
-                // FIXME: This is too simplistic. We have to handle the case of rows
-                // having keys like `expose` set, and ensure the correct configuration
-                // bubbles up
-                return false;
-            }
-            existMap.set(row.id, true);
-            return true;
-        })
-        .sort((a, b) => a.id.localeCompare(b.id));
-}
-
-module.exports = async feeds => {
+const feedEntrypoint = require('./feed-entrypoint');
+const unpackFeed = require('./unpack-feed');
+const { join } = require('path');
+const browserify = require('browserify');
+const getStream = require('get-stream');
+const hashFeed = require('./hash-feed');
+const tmp = require('tmp');
+const { promisify } = require('util');
+const directory = promisify(tmp.dir);
```

Comment:
Can use https://github.com/sindresorhus/tempy, since you're adding a dependency anyways